### ISSUE TITLE
Not creating OS users

### ISSUE THREAD

**Issue by Snowned0425** [cwaldbieser/jhub_remote_user_authenticator#22]
Hello, I have this installed and everything is working great except for one issue. If a new user logs in, it creates the user within JupyterHub, BUT it does not create the underlying OS user. Because of that, I can't run anything as that user and get all sorts of 500 errors. 

Is this supposed to also create the OS user? Do I need some other config to do that?

**Comment by Snowned0425** [comment]
Did some more playing around with this. If we use the "RemoteUserLocalAuthenticator" class, it will create the OS user when the Jupyterhub service is restarted. But it's not creating the OS user on the initial login like it should. So when a user logs in and the account is created, they can't use Jupyter until we restart the service (throws lots of errors if they try to do anything). Is it intentionally waiting for the Jupyterhub service to restart to create the OS user, or is this a bug?

**Comment by cwaldbieser** [comment]
Would you be able to share your Jupyterhub config file with any sensitive information redacted?

**Comment by Snowned0425** [comment]
I think this ended up being a bug in the underlying JupyterHub version. We were using the one shipped with AWS EMR, and it was a year and a half old version (1.1.0 I think). We had the same OS user creation bug with a SAML module we tried too. We then did a standalone build of 1.4.1 and the SAML bug was resolved, so presumably it was for this too. We went with SAML because it's our standard whenever we can.
